dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/JarLinkageReport.java

```diff
@@ -88,11 +91,16 @@ public String toString() {
     }
     return builder.toString();
   }
-
+  
   /** Returns map from the cause of linkage errors to class names affected by the errors. */
   // TODO this is only used by formatJarLinkageReport in macros.ftl. We should be able to
   // refactor this to make it a lot clearer by removing ImmutableMultimap from the API.
-  public ImmutableMultimap<LinkageErrorCause, String> getCauseToSourceClasses() {
+  @Memoized
+  ImmutableMultimap<LinkageErrorCause, String> getCauseToSourceClasses() {
+    return mapCauseToSourceClasses();
```

Comment:
You can inline the private method now, reverting all lines except the new `@Memoized` line (and maybe keeping the default access).